probe-rs/src/bin/probe-rs/cmd/run.rs

```diff
@@ -91,20 +97,32 @@ impl Cmd {
                 loader,
                 self.chip_erase,
             )?;
+            // reset the core to leave it in a consistent state after flashing
```

Comment:
For the stm32f2 chip I'm using this is required, though I'm not sure why. Possibly a flash loader issue? Either way,I don't think its such a bad thing to reset after flashing, at least then you're guaranteed to be in a consistent state.

Reply:
Resetting after flashing is totally fine, I'm only worried about resetting without flashing first, ie. the `attach` use case.